gcp-observability/src/main/java/io/grpc/gcp/observability/interceptors/ConditionalClientInterceptor.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2022 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.gcp.observability.interceptors;
```

Comment:
Could this live in grpc.io and be useful in other situations? Or is this very observability specific?

Reply:
You mean in some common place like `io.grpc.ClientInterceptors` ? I did think about that. CC @ejona86 : a class like this could be useful as a conditional delegating interceptor

Reply:
Honestly, it isn't obvious to me that the indirection and creation of concepts provides any real gain.

Compare the current calling code:

```java
    return new ConditionalClientInterceptor(interceptor,
        (m, c) -> !SERVICES_TO_EXCLUDE.contains(m.getServiceName()));
```

To just doing it 100% manually:
```java
    return (m, c, n) ->
      SERVICES_TO_EXCLUDE.contains(m.getServiceName()))) ?
        n.newCall(m, c) : interceptor.interceptCall(m, c, n);
    };
```

And that doesn't have the risk of inverting `predicate` because it isn't documented whether true means "use interceptor" or "don't use interceptor".